### ISSUE TITLE
Make Yaktor DSL file extensions consistent

### ISSUE THREAD

**Issue by matthewadams** [SciSpike/yaktor#25]
Currently, domain model DSL files use `.dm`, and conversation DSL files use `.cl` for their extensions.  Being type A, these have always bugged me because they are inconsistent with one another.  They should either both use `l` (for "language") or neither should, becoming either `.dml` & `.cl`, or they should be `.dm` & `.c`.

We obviously don't want to pwn `.c`, so `.dm` & `.c` are out.

I might recommend `.dml` & `.cl`, but I think I'd rather that we choose `.ydm` & `.yc`, for "Yaktor Domain Model" & "Yaktor Conversation", respectively, which are much more unique while still short.

**Comment by barrettc** [comment]
I vote .ydm & .yc

**Comment by matthewadams** [comment]
I vote `.ydm` & `.yc`

**Comment by matthewadams** [comment]
After I had a discussion with @gacord23 & after @jkamke reviewed the PRs for this issue with me, @jkamke & I decided that there are some other changes necessary.

* Change YAPI DSL keyword `conversation` to `api` & change proposed file extension from `.yc` to `.yapi`

The keyword `conversation` is a misnomer.

First, it's a namespace.  We have often said that a conversation is between agents concerning some data.  Thing is, you can import agents from other conversations, meaning that we're kind of mixing agents across conversations.

Second, it's an API, not just a conversation.  It specifies expected data structures & resources in addition to agents.  The interaction a client has with a conversation is by sending messages to agents as well as making requests of resources.  It makes more sense to us to call the collection of data structures, agents & resources an API.

Further, since a `.ydm` file contains a domain model (with the first token being `domain-model`), symmetry dictates that the first token in a `.yapi` file should be `api`.

* Change YAPI DSL keyword `type` to `message`

The keyword `type` is overloaded between the Domain & YAPI DSLs, and they are subtly different, leading to confusion.  We won't change `type` in the Domain DSL.  We had considered changing `type` in the YAPI DSL to `dto`, but were never really happy with that except that it was short, somewhat reflected its purpose, and that it was different from Domain's `type`.  After looking at [Google Protocol Buffers (GPBs)](https://developers.google.com/protocol-buffers/), we liked that they use `message`, so we decided to go with that, because, after all, you do send ``message``s to ``agent``s.

We further recognized that, since our DSL supports projections via the `from` keyword, our capabilities are a superset of GPBs.  However, we could integrate with GPBs by generating GPB `.proto` files after projection for the user (or us) to compile to a particular language binding.  This allows us to communicate messages over the wire efficiently & portably in binary form.

**Comment by matthewadams** [comment]
After sleeping on it & discussing with @jkamke, we came to the conclusion that we're willing to make the compromise of redefining `conversation` to mean "an API that includes messages, agents and resources" rather than change absolutely everything (directories, namespaces, generator code, documentation & how we talk about these things).  We'll keep with the `.yc` extension.

We also discussed renaming `resource` to `endpoint` to make it more obvious, but that discussion will be in another issue.

We're still on board with changing `type` to `message`, though, as `message` succinctly captures the essence of what it actually is and it is different from the Domain DSL's `type`.
